File: python/algorithms/factorial_recursive.py
"""팩토리얼 (재귀) 예시."""


def factorial_recursive(n: int) -> int:
    if n < 0:
        raise ValueError("n은 0 이상이어야 합니다.")
    if n <= 1:
        return 1
    return n * factorial_recursive(n - 1)


if __name__ == "__main__":
    print(factorial_recursive(5))  # 120

File: javascript/algorithms/factorial_recursive.js
function factorialRecursive(n) {
  if (n < 0) throw new Error('n은 0 이상이어야 합니다.');
  if (n <= 1) return 1;
  return n * factorialRecursive(n - 1);
}

console.log(factorialRecursive(5));
